IPython/utils/text.py

```diff
@@ -561,7 +561,13 @@ def _vformat(self, format_string, args, kwargs, used_args, recursion_depth):
 
                 # eval the contents of the field for the object
                 # to be formatted
-                obj = eval(field_name, kwargs)
+                try:
+                    obj = eval(field_name, kwargs)
+                except NameError:
+                    if supressExceptions:
+                       obj = field_name
```

Comment:
We definitely shouldn't change `Formatter.parse()`, that's part of the stdlib.

I was in favour of this because I thought it was going to be a very simple change. It now looks like it's rather more complicated, and I don't think it's worth the trouble. I don't think users very often want a literal `$foo` on the same line as interpolating a Python variable `$bar`.